src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -89,9 +152,42 @@ public String listOwned() {
     @Timed
     @UnitOfWork
     @ApiOperation(value = "Refresh repos owned by the logged-in user", notes = "This part needs to be fleshed out but the user "
-            + "can trigger a sync on the repos they're associated with", response = String.class)
-    public String refreshOwned() {
-        throw new UnsupportedOperationException();
+            + "can trigger a sync on the repos they're associated with", response = Container.class)
+    public List<Container> refreshRepos(@QueryParam("user_id") Long userId) {
+        List<Container> currentRepos = containerDAO.findByUserId(userId);
+        List<Container> allRepos = new ArrayList<>(0);
+        List<Token> tokens = tokenDAO.findByUserId(userId);
+
+        for (Token token : tokens) {
+            if (token.getTokenSource().equals(TokenType.QUAY_IO.toString())) {
+                Optional<String> asString = ResourceUtilities.asString(TARGET_URL + "repository?last_modified=true&public=false",
+                        token.getContent(), client);
+
+                if (asString.isPresent()) {
+                    RepoList repos;
+                    try {
+                        repos = MAPPER.readValue(asString.get(), RepoList.class);
+                        allRepos.addAll(repos.getRepositories());
+                    } catch (IOException ex) {
+                        Logger.getLogger(DockerRepoResource.class.getName()).log(Level.SEVERE, null, ex);
+                    }
+                }
+            }
+        }
+
+        for (Container newContainer : allRepos) {
+            for (Container oldContainer : currentRepos) {
+                if (newContainer.getName().equals(oldContainer.getName())
+                        && newContainer.getNamespace().equals(oldContainer.getNamespace())) {
+                    System.out.println("container " + oldContainer.getId() + " is being updated ...");
```

Comment:
There's a mix of Logger (from line 172) and System.out here. Note that dropwizard uses Logback, so just standardize on that. http://logback.qos.ch/manual/configuration.html has examples